hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -516,6 +519,16 @@ hasPT headCurrencySymbol txOut =
    in length pts == 1
 {-# INLINEABLE hasPT #-}
 
+mustNotBurnTokens :: ScriptContext -> CurrencySymbol -> Bool
+mustNotBurnTokens ScriptContext{scriptContextTxInfo} headCurrencySymbol =
+  case Map.lookup headCurrencySymbol minted of
+    Nothing -> True
+    Just tokenMap ->
+      traceIfFalse "burning is forbidden" $ Map.empty == tokenMap
+ where
+  minted = getValue (txInfoMint scriptContextTxInfo)
+{-# INLINEABLE mustNotBurnTokens #-}
```

Comment:
The specification currently forbids **any** minting or burning. This implementation here only checks the minting of our currency id.

Must: change to disallow any minting or burning.